### PR TITLE
feat: add export_file_to_dataframe function 

### PR DESCRIPTION
**Pull request by manesioz**
Allows user to export _entire_ table to a Pandas dataframe, regardless of size

### CODE REVIEW

**Comment by manesioz** [comment]
cc. @brendanstennett

File: namara/__init__.py [hunk 1]
```diff
@@ -10,7 +13,23 @@ def __init__(self, api_key, debug=False, host='https://api.namara.io', api_versi
         self.api_version = api_version
         self.base_path = '{0}/{1}'.format(self.host, self.api_version)
         self.headers = {'Content-Type': 'application/json', 'X-API-Key': api_key}
-
+    
+    def export_file_to_dataframe(self, dataset_id, organization_id, project_id, options=None):
```
Comment (brendanstennett) [inline namara/__init__.py:17]: Name this `export`.  Include `output_format='url'` as default.

File: namara/__init__.py [hunk 2]
```diff
@@ -10,7 +13,23 @@ def __init__(self, api_key, debug=False, host='https://api.namara.io', api_versi
...
+        list_of_chunks = [] 
+        url = self.get_url(f'/data_sets/{dataset_id}/data/export?geometry_format=wkt&organization_id={organization_id}&project_id={project_id}')
+        while True:
+            response = self.__session.get(url, params=options, headers=self.headers).result().json()
+            if self.debug: 
+                logging.debug('REQUEST URL: ' + response['url'])
+            if response['message'] == 'Exported':
+                for chunk in pd.read_csv(response['url'], chunksize=100):
```
Comment (brendanstennett) [inline namara/__init__.py:25]: If `output_format == 'url'` return the URL
If `output_format == 'dataframe'` do this.

File: test_namara.py [hunk 3]
```diff
@@ -42,7 +43,7 @@ def test_valid_json_output(self):
         response = self.subject.get(self.dataset, self.version, options={'operation': 'count(*)'}, output_format='json')
         self.assertTrue(isinstance(response, list))
 
-    def test_valid_dataframe_ouput(self): 
+    def test_valid_get_dataframe_ouput(self):
```
Comment (brendanstennett) [inline test_namara.py:46]: Should include tests to the export function as well.

File: namara/__init__.py [hunk 4]
```diff
@@ -14,21 +15,24 @@ def __init__(self, api_key, debug=False, host='https://api.namara.io', api_versi
         self.base_path = '{0}/{1}'.format(self.host, self.api_version)
         self.headers = {'Content-Type': 'application/json', 'X-API-Key': api_key}
     
-    def export_file_to_dataframe(self, dataset_id, organization_id, project_id, options=None):
-        list_of_chunks = [] 
+    @retry(wait=wait.wait_exponential(multiplier=1, min=4, max=10), stop=stop.stop_after_attempt(5))
```
Comment (brendanstennett) [inline namara/__init__.py:18]: Retry isn't necessary and in fact could lead to some other issues.  I would just get rid of this.

File: test_namara.py [hunk 5]
```diff
@@ -24,7 +26,9 @@ def test_is_aggregation(self):
...
         self.subject.get = Mock(return_value=[{u'mnr_region': u'NORTHWEST', u'facility_name': u'RESOLUTE FP CANADA INC.', u'facility_code': 1201, u'location': u'FORT FRANCES', u'facility_type': u'PULP'}, {u'mnr_region': u'NORTHWEST', u'facility_name': u'MANITOU FOREST PRODUCTS LTD.', u'facility_code': 1221, u'location': u'EMO', u'facility_type': u'SAWMILL'}, {u'mnr_region': u'NORTHWEST', u'facility_name': u'531322 ONTARIO LTD. O/A NICKEL LAKE LUMBER', u'facility_code': 1232, u'location': u'FORT FRANCES', u'facility_type': u'SAWMILL'}, {u'mnr_region': u'NORTHWEST', u'facility_name': u'AINSWORTH GP LTD.', u'facility_code': 1240, u'location': u'BARWICK', u'facility_type': u'COMPOSITE'}, {u'mnr_region': u'NORTHWEST', u'facility_name': u'RESOLUTE FP CANADA INC.', u'facility_code': 1301, u'location': u'IGNACE', u'facility_type': u'SAWMILL'}, {u'mnr_region': u'NORTHWEST', u'facility_name': u'E.&G. CUSTOM SAWING LTD.', u'facility_code': 1410, u'location': u'KENORA', u'facility_type': u'SAWMILL'}, {u'mnr_region': u'NORTHWEST', u'facility_name': u'WEYERHAEUSER COMPANY LTD.', u'facility_code': 1422, u'location': u'KENORA', u'facility_type': u'COMPOSITE'}])
+        # print(self.subject.get)
```
Comment (brendanstennett) [inline test_namara.py]: Remove

File: test_namara.py [hunk 6]
```diff
@@ -24,7 +26,9 @@ def test_is_aggregation(self):
...
         response = self.subject.get(self.dataset, self.version, options={'where': 'facility_code > 1200'})
+        # print(response)
```
Comment (brendanstennett) [inline test_namara.py]: Remove

File: namara/__init__.py [hunk 7]
```diff
@@ -10,7 +15,34 @@ def __init__(self, api_key, debug=False, host='https://api.namara.io', api_versi
...
+                    return pd.concat(list_of_chunks)
+                elif output_format == 'csv': 
+                    with requests.get(response['url'], stream=True) as r:
+                        shutil.copyfileobj(r.raw, output_file)
+                    return
+                else: 
+                    raise Exception('`output_format` param must be "csv", "dataframe", or "url" (default)')
+            elif 'message' in response and response['message'] == 'Failed':
+                raise Exception(f"Could not export dataset {dataset_id}")
+            elif 'message' in response and response['message'] == 'Pending':
```
Comment (brendanstennett) [inline namara/__init__.py:41]: I think "Exporting" is a potential state as well.

Comment (manesioz) [reply namara/__init__.py:41]: Good catch

**Pull request merged**